lib/realtime.nit

```diff
@@ -142,18 +163,43 @@ class Clock
 		return tv;
 	`}
 
-	# Return timelapse since instanciation of this instance
-	fun total : Timespec
+	# Seconds since the creation of this instance
+	fun total: Float
 	do
-		return new Timespec.monotonic_now - time_at_beginning
+		var now = new Timespec.monotonic_now
+		var diff = now - time_at_beginning
+		var r = diff.to_f
+		diff.free
+		now.free
+		return r
 	end
 
-	# Return timelapse since last call to lapse
-	fun lapse : Timespec
+	# Seconds since the last call to `lapse`
+	fun lapse: Float
 	do
 		var nt = new Timespec.monotonic_now
 		var dt = nt - time_at_last_lapse
+		var r = dt.to_f
+		dt.free
+		time_at_last_lapse.free
 		time_at_last_lapse = nt
-		return dt
+		return r
+	end
+
+	# Seconds since the last call to `lapse`, without resetting the lapse counter
+	fun peek_lapse: Float
+	do
+		var nt = new Timespec.monotonic_now
+		var dt = nt - time_at_last_lapse
+		var r = dt.to_f
+		nt.free
+		dt.free
+		return r
+	end
+
+	redef fun finalize_once
+	do
+		time_at_beginning.free
+		time_at_last_lapse.free
```

Comment:
The attributes are already `protected` and they are not leaked, we should be fine.

But yes, when using the finalization to free attributes, respecting the encapsulation is critical!